atom/browser/api/atom_api_web_contents.cc

```diff
@@ -425,6 +427,23 @@ void WebContents::InitZoomController(content::WebContents* web_contents,
     zoom_controller_->SetDefaultZoomFactor(zoom_factor);
 }
 
+#if defined(OS_WIN) || defined(OS_LINUX)
+void WebContents::SetAcceptLanguages(content::RendererPreferences* prefs) {
+#ifdef OS_WIN
+  std::vector<base::string16> languages;
+  base::win::i18n::GetThreadPreferredUILanguageList(&languages);
+  std::string accept_langs;
+  for (const auto& s16 : languages) {
+    accept_langs += base::SysWideToUTF8(s16) + ',';
+  }
+  accept_langs.pop_back();
+  prefs->accept_languages = accept_langs;
+#else  // OS_LINUX
+  prefs->accept_languages = g_browser_process->GetApplicationLocale();
+#endif
+}
+#endif
```

Comment:
Too many `#ifdef`s! Perhaps;

```cpp
void WebContents::SetAcceptLanguages(content::RendererPreferences* prefs) {
#if defined(OS_WIN) || defined(OS_LINUX)
  prefs->accept_languages = GetPreferredOSLanguages();
#else
  NOTREACHED(); // or... what happens on mac?
#endif
}
```
and then define `GetPreferredOSLanguages` in a separate file, possibly using the `foo.h` / `foo_win.cc` / `foo_linux.cc` pattern?

Reply:
Yes, I realised that. We can create separate files.